packages/registry/src/registry.test.tsx

```diff
@@ -0,0 +1,374 @@
+    );
+  });
+  it('registers and gets a component by name (as an object)', () => {
+    config.registerComponent({
+      name: 'Toolbar.Bar',
+      component: 'this is a Bar component',
+    });
+    expect(config.getComponent({ name: 'Toolbar.Bar' }).component).toEqual(
+      'this is a Bar component',
+    );
+  });
+  it('registers and gets a component by name (as an object) - check displayName', () => {
+    config.registerComponent({
+      name: 'Toolbar.Bar',
+      component: () => <div>Hello</div>,
+    });
+    expect(config.getComponent('Toolbar.Bar').component.displayName).toEqual(
+      'Toolbar.Bar',
+    );
+  });
+  it('registers and gets a component by name (as an object) - check displayName if it has already one, it does not overwrite it', () => {
+    const TestComponent = () => <div>Hello</div>;
+    TestComponent.displayName = 'DisplayNameAlreadySet';
+    config.registerComponent({
+      name: 'Toolbar.Bar',
+      component: TestComponent,
+    });
+
+    expect(config.getComponent('Toolbar.Bar').component.displayName).toEqual(
+      'DisplayNameAlreadySet',
+    );
+  });
+  it('registers and gets a component by name (as an object) - check displayName - do not break if it is a normal function', () => {
+    function myFunction() {
+      return 'true';
+    }
+
+    config.registerComponent({
+      name: 'Toolbar.Bar',
+      component: myFunction,
+    });
+    expect(config.getComponent('Toolbar.Bar').component.displayName).toEqual(
+      'Toolbar.Bar',
+    );
+  });
+  it('registers a component by name with dependencies', () => {
+    config.registerComponent({
+      name: 'Toolbar.Bar',
+      component: 'this is a Bar component',
+      dependencies: 'News Item',
+    });
+    expect(
+      config.getComponent({ name: 'Toolbar.Bar', dependencies: 'News Item' })
+        .component,
+    ).toEqual('this is a Bar component');
+  });
+  it('registers a component by name with dependencies array', () => {
+    config.registerComponent({
+      name: 'Toolbar.Bar',
+      component: 'this is a Bar component',
+      dependencies: ['News Item', 'StringFieldWidget'],
+    });
+    expect(
+      config.getComponent({
+        name: 'Toolbar.Bar',
+        dependencies: ['News Item', 'StringFieldWidget'],
+      }).component,
+    ).toEqual('this is a Bar component');
+  });
+});
+
+describe('Slots registry', () => {
+  // config.slots.toolbar = [ // viewlets.xml
+  //   'save',
+  //   'edit',
+  // ]
+
+  // config.slots.toolbar.save = [
+  //   {
+  //     predicates: [RouteCondition('/de')],
+  //     component: 'this is a Bar component',
+  //   },
+  //   {
+  //     predicates: [RouteCondition('/de'), ContentTypeCondition(['News Item'])],
+  //     component: 'this is a Bar component',
+  //   },
+  //   {
+  //     predicates: [RouteCondition('/de/path/folder'), ContentTypeCondition(['News Item']), TrueCondition()],
+  //     component: 'this is a Bar component',
+  //   },
+  // ]
```

Comment:
Lots of commented code here. Is this necessary?

Reply:
Removed